File: 1201/4811.js
const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});
const dp = new Array(31).fill().map((e) => Array(31).fill(0));
const go = (W, H) => {
  // W: 한 개 알약. H: 반 개 알약 각 개수.
  if (W === 0 && H === 0) return 1; // 가장 마지막 끝이 되는 부분에서 1을 반환해서 차곡차곡 경우의 수를 더해나간다.
  if (dp[W][H]) return dp[W][H];

  if (W > 0) {
    // 한 개 짜리 알약을 꺼낸다면, 반개는 새로 생기고, 한 개는 개수가 줄어듦.
    dp[W][H] += go(W - 1, H + 1);
  }
  if (H > 0) {
    // 반 개 짜리 알약을 꺼낸다면
    dp[W][H] += go(W, H - 1); // 반 개는 개수만 줄어듦.
  }
  return dp[W][H]; // 합쳐진 경우의 수를 반환.
};

rl.on("line", (line) => {
  if (+line === 0) {
    rl.close();
  } else {
    console.log(go(+line, 0));
  }
});

rl.on("close", () => {
  process.exit();
});
//  백준 알약

File: 0105/4811.py
dp= [[0]*31 for _ in range(31)]# dp를 이용해서 900번 연산으로 값을 구할 수 있다.
def go(W,H):# H와 W의 개수만 알면 구할 수 있음. dp[w][h] = dp[w-1][h+1] + dp[w][h-1].
    if W == 0 and H == 0:# 모두 없어지면 하나의 경우이므로 1반환.
        return 1
    if dp[W][H]:
        return dp[W][H]
    if W > 0:
        dp[W][H] += go(W-1,H+1)
    if H > 0:
        dp[W][H] += go(W,H-1)
    return dp[W][H]

while True:
    n = int(input())
    if n == 0:
        break
    print(go(n,0))
# 백준 알약 문제